### PR TITLE
Windows: Retry convertToPlaceholder for 30s on new dir

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/libsync/propagateremotemkdir.cpp [hunk 1]
```diff
@@ -131,15 +134,57 @@ void PropagateRemoteMkdir::success()
     itemCopy._etag.clear();
 
     // save the file id already so we can detect rename or remove
```
Comment (dragotin) [inline src/libsync/propagateremotemkdir.cpp:136]: This comment does not really look useful.

File: src/libsync/propagateremotemkdir.cpp [hunk 2]
```diff
@@ -131,15 +134,57 @@ void PropagateRemoteMkdir::success()
...
+                done(SyncFileItem::Success);
+            } else {
+                done(SyncFileItem::SoftError, result->error());
+            }
+        });
+
+        watcher->setFuture(QtConcurrent::run([p = propagator(), itemCopy] {
+            // Try to update the meta data with a 30s timeout
+            const auto start = std::chrono::steady_clock::now();
+            while ((std::chrono::steady_clock::now() - start) < UpdateMetaDataRetyTimeOut) {
```
Comment (dragotin) [inline src/libsync/propagateremotemkdir.cpp:170]: I like `QElapsedTimer` for these kind of things.

File: src/libsync/propagateremotemkdir.cpp [hunk 3]
```diff
@@ -131,15 +134,57 @@ void PropagateRemoteMkdir::success()
...
+                QThread::sleep(1);
```
Comment (dragotin) [inline src/libsync/propagateremotemkdir.cpp:171]: Any reason why you sleep a second before trying the first time?

**Review by dragotin — APPROVED** [review]
Ok, minor comments. Except the sleep before trying the first time.

File: src/libsync/propagateremotemkdir.cpp (hunk 2 above)
Comment (TheOneRing) [reply src/libsync/propagateremotemkdir.cpp:170]: Hm but then we have to either convert units or use std::chrono for the conversion and in the end we don't gain much  besides more involved components.

File: src/libsync/propagateremotemkdir.cpp (hunk 3 above)
Comment (TheOneRing) [reply src/libsync/propagateremotemkdir.cpp:172]: The idea is that we really just tried that right before so no need to retry.

**Pull request merged**